plugins/groovy/groovy-psi/src/org/jetbrains/plugins/groovy/findUsages/GradlePropertyReferencesSearcher.java

```diff
@@ -0,0 +1,83 @@
+// Copyright 2000-2019 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE file.
+package org.jetbrains.plugins.groovy.findUsages;
+
+import com.intellij.lang.properties.psi.Property;
+import com.intellij.openapi.application.QueryExecutorBase;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiReference;
+import com.intellij.psi.search.DelegatingGlobalSearchScope;
+import com.intellij.psi.search.GlobalSearchScope;
+import com.intellij.psi.search.RequestResultProcessor;
+import com.intellij.psi.search.UsageSearchContext;
+import com.intellij.psi.search.searches.ReferencesSearch;
+import com.intellij.util.Processor;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.plugins.groovy.lang.psi.api.statements.expressions.GrReferenceExpression;
```

Comment:
this is the gist of the problem discussed in the e-mail. If I try to "Refactor->Move" this class into the Gradle plugin (where, as I feel, it belongs), it says "I cannot import this symbol anymore when this class is in the other module" (which is pretty obvious now that I think of it). So, how can I use symbols (types, methods, etc) from plugin A in a source of plugin B? Or what is the proper way to dance around this problem ? @ForNeVeR

Reply:
I think I can probably just use the PsiReference interface in my checks and type-casting; I'll check if it works. thanks

Reply:
and now the ReferenceSearch is moved into Gradle plugin, where it belongs!